src/views/clusteroverview/SettingsTab/ClusterTab/components/ResourceManagementSection/components/AutoComputeCPULimits/ProjectSelectorModal.tsx

```diff
@@ -0,0 +1,146 @@
+import React, { Dispatch, FC, SetStateAction } from 'react';
+
+import { modelToGroupVersionKind, ProjectModel } from '@kubevirt-ui/kubevirt-api/console';
+import LabelList from '@kubevirt-utils/components/NodeSelectorModal/components/LabelList';
+import LabelRow from '@kubevirt-utils/components/NodeSelectorModal/components/LabelRow';
+import { useIDEntities } from '@kubevirt-utils/components/NodeSelectorModal/hooks/useIDEntities';
+import { IDLabel } from '@kubevirt-utils/components/NodeSelectorModal/utils/types';
+import ProjectCheckerAlert from '@kubevirt-utils/components/ProjectCheckerAlert/ProjectCheckerAlert';
+import SimpleCopyButton from '@kubevirt-utils/components/SimpleCopyButton/SimpleCopyButton';
+import { useKubevirtTranslation } from '@kubevirt-utils/hooks/useKubevirtTranslation';
+import { K8sResourceKind, useK8sWatchResource } from '@openshift-console/dynamic-plugin-sdk';
+import {
+  ActionList,
+  ActionListItem,
+  Button,
+  ButtonVariant,
+  Form,
+  Label,
+  Modal,
+  ModalVariant,
+  Stack,
+  StackItem,
+  Text,
+} from '@patternfly/react-core';
+
+import { projectSelectorToIDLabels } from '../../utils/utils';
+
+import './ProjectSelectorModal.scss';
+
+type ProjectSelectorModalProps = {
+  isOpen: boolean;
+  labels?: { [key: string]: string };
+  onClose: () => void;
+  onSubmit: Dispatch<SetStateAction<any[]>>;
+};
+
+const ProjectSelectorModal: FC<ProjectSelectorModalProps> = ({
+  isOpen,
+  labels,
+  onClose,
+  onSubmit,
+}) => {
+  const { t } = useKubevirtTranslation();
+  const {
+    entities: selectorLabels,
+    onEntityAdd: onLabelAdd,
+    onEntityChange: onLabelChange,
+    onEntityDelete: onLabelDelete,
+  } = useIDEntities<IDLabel>(projectSelectorToIDLabels(labels));
+
+  const [projects, projectsLoaded] = useK8sWatchResource<K8sResourceKind[]>({
+    groupVersionKind: modelToGroupVersionKind(ProjectModel),
+    isList: true,
+  });
+
+  const qualifiedProjects = projects.filter((project) =>
+    selectorLabels.every((label) => project?.metadata?.labels[label.key] === label.value),
+  );
+
+  const onSelectorLabelAdd = () => onLabelAdd({ id: null, key: '', value: '' });
+
+  const handleSubmit = () => {
+    onSubmit(selectorLabels);
+    onClose();
+  };
+
+  return (
+    <Modal
+      className="ocs-modal project-selector-modal"
+      isOpen={isOpen}
+      onClose={onClose}
+      title={t('Project selector')}
+      variant={ModalVariant.small}
+    >
+      <Stack hasGutter>
+        <StackItem>
+          <Text className="project-selector-modal__header-text">
+            {t('Automatically compute CPU limits on projects containing these labels')}
+          </Text>
+        </StackItem>
+        <StackItem>
+          <Label
+            onClick={(e) => {
+              e.preventDefault();
+            }}
+            color="grey"
+            href="#"
```

Comment:
?

Reply:
I used `NodeSelectorModal` as the start for this component. This code is there as well. I deleted it and it didn't seem to make a difference.